### PR TITLE
Change all null checking expressions to use 'is' and 'is not'.

### PR DESCRIPTION
**Pull request by rolfbjarne**
Change all null checking expressions to use 'is null' and 'is not null'
instead of '== null' and '!= null'.

This was mostly done with sed, so code can probably be improved in many other
ways with manual inspection, but that will come over time.

### CODE REVIEW

**Review by haritha-mohan — APPROVED** [review]

File: tools/common/Driver.execution.cs [hunk 1]
```diff
@@ -70,8 +70,8 @@ public static int RunCommand (string path, IList<string> args, StringBuilder out
 
 		public static int RunCommand (string path, IList<string> args, Dictionary<string, string> env, StringBuilder output, StringBuilder error, bool suppressPrintOnErrors, int verbosity)
 		{
-			var output_received = output == null ? null : new Action<string> ((v) => { if (v != null) output.AppendLine (v); });
-			var error_received = error == null ? null : new Action<string> ((v) => { if (v != null) error.AppendLine (v); });
+			var output_received = output is null ? null : new Action<string> ((v) => { if (v is not null) output.AppendLine (v); });
+			var error_received = error is null ? null : new Action<string> ((v) => { if (v is not null) error.AppendLine (v); });
```
Comment (tj-devel709) [inline tools/common/Driver.execution.cs:74]:
```suggestion
			var output_received = output ?? new Action<string> ((v) => { if (v is not null) output.AppendLine (v); });
			var error_received = error ?? new Action<string> ((v) => { if (v is not null) error.AppendLine (v); });
```

**Review by tj-devel709 — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Review by mauroa — APPROVED** [review]

**Review by emaf — APPROVED** [review]

**Review by stephen-hawley — COMMENTED** [review]
This is a big PR - taking it in pieces.

**Review by stephen-hawley — CHANGES_REQUESTED** [review]
I know this is a pain, but since it looks like we're committed to nullable, I'd like to see a test that greps through all the source code for `== null`, `!= null`, `null ==`, and `null !=` just so that we keep ourselves honest.

**Comment by rolfbjarne** [comment]
@stephen-hawley 

> I know this is a pain, but since it looks like we're committed to nullable, I'd like to see a test that greps through all the source code for `== null`, `!= null`, `null ==`, and `null !=` just so that we keep ourselves honest.

I made it better than a test: automatically fix it (using autoformat): [xamarin/xamarin-macios@`2f91cd0` (#18176)](https://github.com/xamarin/xamarin-macios/pull/18176/commits/2f91cd0875e4b062e2a32e7e55bd0b0ebf6667cd)

Note I only handle the two first cases (null as the second operator), because:

* It's much more complex to handle null as the first operator, because you have to switch the order (`null == x` => `x is null`)
* It's a very rare thing to do in our code base (I found it in two files: one is fixed, and the other is actually correct because it's a test testing equality operators and the code shouldn't be changed).

**Review by stephen-hawley — APPROVED** [review]

**Pull request merged**